src/main/java/ca/corefacility/bioinformatics/irida/processing/concatenate/SequencingObjectConcatenator.java

```diff
@@ -1,63 +1,97 @@
-
-import org.apache.commons.io.IOUtils;
-
 import ca.corefacility.bioinformatics.irida.exceptions.ConcatenateException;
 import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
 import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Optional;
 
 /**
- * Class to concatenate multiple {@link SequencingObject}s and return a single
- * new {@link SequencingObject}. This class should be extended by
- * implementations for specific {@link SequencingObject}s
- * 
- * @param <Type>
- *            the {@link SequencingObject} class to concatenate
+ * Class to concatenate multiple {@link SequencingObject}s and return a single new {@link SequencingObject}. This class
+ * should be extended by implementations for specific {@link SequencingObject}s
+ *
+ * @param <Type> the {@link SequencingObject} class to concatenate
  */
 public abstract class SequencingObjectConcatenator<Type extends SequencingObject> {
+	private static final List<String> VALID_EXTENSIONS = Lists.newArrayList("fastq", "fastq.gz");
 
 	/**
 	 * Concatenate a set of {@link SequencingObject}s of a given type
-	 * 
-	 * @param toConcatenate
-	 *            the set of {@link SequencingObject}s to concatenate
-	 * @param filename
-	 *            base name of the new file to create
+	 *
+	 * @param toConcatenate the set of {@link SequencingObject}s to concatenate
+	 * @param filename      base name of the new file to create
 	 * @return the newly created {@link SequencingObject} class
-	 * @throws ConcatenateException
-	 *             if there is an error during concatenation
+	 * @throws ConcatenateException if there is an error during concatenation
 	 */
 	public abstract Type concatenateFiles(List<? extends SequencingObject> toConcatenate, String filename)
 			throws ConcatenateException;
 
 	/**
 	 * Append a {@link SequenceFile} to a {@link Path} on the filesystem
-	 * 
-	 * @param target
-	 *            the {@link Path} to append to
-	 * @param file
-	 *            the {@link SequenceFile} to append to the path
-	 * @throws ConcatenateException
-	 *             if there is an error appending the file
+	 *
+	 * @param target the {@link Path} to append to
+	 * @param file   the {@link SequenceFile} to append to the path
+	 * @throws ConcatenateException if there is an error appending the file
 	 */
 	protected void appendToFile(Path target, SequenceFile file) throws ConcatenateException {
 
-		try (FileWriter fw = new FileWriter(target.toFile(), true);
-				BufferedWriter writer = new BufferedWriter(fw);
-				FileReader reader = new FileReader(file.getFile().toFile())) {
-
-			IOUtils.copy(reader, writer);
+		try (FileChannel out = FileChannel.open(target, StandardOpenOption.CREATE, StandardOpenOption.APPEND,
```

Comment:
Switched this to use `FileChannel` instead of the `FileWriter`.  `FileWriter` was having issues reading/writing the gzip files.  `FileChannel` is also a fair bit faster.